### PR TITLE
Boards and Tagger stuff

### PR DESCRIPTION
**Pull request by SJuliez**
Upates the BoardsTagger with the (obvious) improvement of not saving a board file when the autotags dont change to avoid having it flagged as changed.
Also updates some boards of AGoAC and Tukayyid. 
- The desert maps were not actually deserts due to the change of default theme
- A few minor hex corrections
- Added theme transitions (intermediate coloring) to make the maps look more like the originals (as much that is possible)
And updates the Boardview to place shadows above ground_fluff (= theme transitions)
AGoAC Desert 2
AGoAC Desert 2
AGoAC Desert 3
AGoAC Desert 3
AGoAC Grasslands 2
AGoAC Grasslands 2
AGoAC Grasslands 3
AGoAC Grasslands 3
Tukayyid Racice
Racice
Tukayyid Deployment Zone
Deployment Zone

### CODE REVIEW

File: megamek/src/megamek/utils/BoardsTagger.java [hunk 1]
```diff
@@ -267,26 +268,35 @@ private static void tagBoard(File boardFile) {
...
-            board.save(os);
-        } catch (IOException e) {
-            System.out.println("Error: Could not save board: " + boardFile);
-        }
+            // Re-save the board
+            try (OutputStream os = new FileOutputStream(boardFile)) {
+                board.save(os);
+            } catch (IOException e) {
+                System.out.println("Error: Could not save board: " + boardFile);
+            }
```
Comment (sixlettervariables) [inline megamek/src/megamek/utils/BoardsTagger.java:297]: Let's also output the `IOException` to see why.

File: megamek/src/megamek/utils/BoardsTagger.java [hunk 2]
```diff
@@ -267,26 +268,36 @@ private static void tagBoard(File boardFile) {
...
+        Set<String> toAdd = matchingTags.keySet().stream().filter(t -> matchingTags.get(t)).map(Tags::getName).collect(toSet());
 
         if (DEBUG) {
             System.out.println("----- Board: " + boardFile);
-            matchingTags.keySet().stream().forEach(System.out::println);
+            if (toRemove.equals(toAdd)) {
+                System.out.println("No changes");
+            } else {
+                toAdd.stream().forEach(System.out::println);
+            }
```
Comment (Windchild292) [inline megamek/src/megamek/utils/BoardsTagger.java:284]: MegaMek Logger swapover

File: megamek/src/megamek/utils/BoardsTagger.java [hunk 3]
```diff
@@ -267,26 +268,36 @@ private static void tagBoard(File boardFile) {
...
-            board.save(os);
-        } catch (IOException e) {
-            System.out.println("Error: Could not save board: " + boardFile);
-        }
+            // Re-save the board
+            try (OutputStream os = new FileOutputStream(boardFile)) {
+                board.save(os);
+            } catch (IOException e) {
+                System.out.println("Error: Could not save board: " + boardFile);
+                e.printStackTrace();
```
Comment (Windchild292) [inline megamek/src/megamek/utils/BoardsTagger.java:297]: MegaMek Logger swapover

**Review by Windchild292 — CHANGES_REQUESTED** [review]

File: megamek/src/megamek/utils/BoardsTagger.java (hunk 2 above)
Comment (SJuliez) [reply megamek/src/megamek/utils/BoardsTagger.java:289]: The tagger is a standalone tool. Seems to make sense to report to the console directly.

File: megamek/src/megamek/client/ui/swing/boardview/BoardView1.java [hunk 4]
```diff
@@ -2675,15 +2675,16 @@ private void drawHex(Coords c, Graphics boardGraph,
             g.setComposite(svComp);
         }
 
-        // To place roads under the shadow map, supers for hexes with roads
+        // To place roads under the shadow map, some supers 
         // have to be drawn before the shadow map, otherwise the supers are
         // drawn after.  Unfortunately I dont think the supers images
```
[the remark this answers is not in the document]
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/boardview/BoardView1.java]: Np. But it's not "I don't think", rather "can't" :)

[the remark this answers is not in the document]
Comment (Windchild292) [reply megamek/src/megamek/client/ui/swing/boardview/BoardView1.java]: I didn't read the comment after the edited line, just noticed the spelling issue while scanning.

File: megamek/src/megamek/utils/BoardsTagger.java (hunk 2 above)
Comment (Windchild292) [reply megamek/src/megamek/utils/BoardsTagger.java:289]: Ah, if this isn't for the logger these can be ignored (I tend to assume logger)

**Review by Windchild292 — APPROVED** [review]

**Pull request merged**